### PR TITLE
Feature/dex 400 dynamic 3DS

### PR DESCRIPTION
**Pull request by EvansPie**
DEX-400

# What this PR does

PoC for the dynamic 3DS flow.

# Notable decisions & other stuff

List & explain ...

# Instructions on how to test this

Use 1 GBP on sandbox. This will return the appropriate response to trigger dynamic 3DS.

# Before merging

_QA_

- [x] I manually tested this with a demo application (simulator)
- [ ] I manually tested this with a demo application (real device)
- [ ] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk

### CODE REVIEW

File: Sources/PrimerSDK/Classes/Core/PCI/TokenizationService.swift [hunk 1]
```diff
@@ -42,18 +45,76 @@ internal class TokenizationService: TokenizationServiceProtocol {
...
+                
+                var isThreeDSEnabled: Bool = false
+                
+                if state.paymentMethodConfig?.paymentMethods?.filter({ ($0.options as? CardOptions)?.threeDSecureEnabled == true }).count ?? 0 > 0 {
+                    isThreeDSEnabled = true
+                }
+                
+                self.tokenizedPaymentMethodToken = paymentMethodToken
+                               
+//                #if canImport(Primer3DS)
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/Core/PCI/TokenizationService.swift:66]: do we want to keep these commented out fields?

File: Sources/PrimerSDK/Classes/Core/Payment Services/VaultService.swift [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #if canImport(UIKit)
 
+import Foundation
 import UIKit
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/Core/Payment Services/VaultService.swift:4]: do we need UIKit here?

File: Tests/PrimerSDK_Tests/ViewControllers/OAuthViewControllerTests.swift [hunk 3]
```diff
@@ -14,33 +14,33 @@ import XCTest
...
-        DependencyContainer.register(router as RouterDelegate)
-        DependencyContainer.register(settings as PrimerSettingsProtocol)
-
-        let viewController = OAuthViewController(host: .paypal)
-
-        viewController.viewDidLoad()
-        
-        wait(for: [expectation], timeout: 10.0)
-    }
+//    func test_paypal_load_exception_shows_error_screen() throws {
```
Comment (ceriksson) [inline Tests/PrimerSDK_Tests/ViewControllers/OAuthViewControllerTests.swift:17]: 🤔

**Review by ceriksson — APPROVED** [review]
Looks good, QA was fine too. Just left some comments regarding commented out fields etc.

File: Sources/PrimerSDK/Classes/Core/PCI/TokenizationService.swift (hunk 1 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Core/PCI/TokenizationService.swift:66]: Fixed

File: Sources/PrimerSDK/Classes/Core/Payment Services/VaultService.swift (hunk 2 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Core/Payment Services/VaultService.swift]: Fixed

**Pull request closed without merging**